plugins/utils/improved-search.nix

```diff
@@ -0,0 +1,122 @@
+# `globalPrefix` argument to avoid the creation of the `settings` option.
+  helpers.vim-plugin.mkVimPlugin config {
+    name = "improved-search";
+    originalName = "improved-search.nvim";
+    defaultPackage = pkgs.vimPlugins.improved-search-nvim;
+
+    maintainers = [maintainers.GaetanLepage];
+
+    extraOptions = {
+      keymaps = mkOption {
+        description = ''
+          Keymap definitions for search functions
+
+          See [here](https://github.com/backdround/improved-search.nvim?tab=readme-ov-file#functions-and-operators) for the list of available callbacks.
+        '';
+        type = with helpers.nixvimTypes;
+          listOf (
+            submodule {
+              options = {
+                key = mkOption {
+                  type = str;
+                  description = "The key to map.";
+                  example = "!";
+                };
+
+                mode = helpers.keymaps.mkModeOption "";
+
+                action = mkOption {
+                  type = with helpers.nixvimTypes;
+                    maybeRaw (
+                      # https://github.com/backdround/improved-search.nvim?tab=readme-ov-file#functions-and-operators
+                      enum [
+                        "stable_next"
+                        "stable_previous"
+                        "current_word"
+                        "current_word_strict"
+                        "in_place"
+                        "in_place_strict"
+                        "forward"
+                        "forward_strict"
+                        "backward"
+                        "backward_strict"
+                      ]
+                    );
+                  description = ''
+                    The action to execute.
+
+                    See [here](https://github.com/backdround/improved-search.nvim?tab=readme-ov-file#functions-and-operators) for the list of available callbacks.
+                  '';
+                  example = "in_place";
+                };
+
+                options = helpers.keymaps.mapConfigOptions;
+              };
+            }
+          );
+        default = [];
+        example = [
+          {
+            mode = ["n" "x" "o"];
+            key = "n";
+            action = "stable_next";
+          }
+          {
+            mode = ["n" "x" "o"];
+            key = "N";
+            action = "stable_previous";
+          }
+          {
+            mode = "n";
+            key = "!";
+            action = "current_word";
+          }
+          {
+            mode = "x";
+            key = "!";
+            action = "in_place";
+          }
+          {
+            mode = "x";
+            key = "*";
+            action = "forward";
+          }
+          {
+            mode = "x";
+            key = "#";
+            action = "backward";
+          }
+          {
+            mode = "n";
+            key = "|";
+            action = "in_place";
```

Comment:
It'd be nice if one or two examples & tests showed using the keymap `options` submodule, e.g. to set `desc`.

Reply:
I added such an example.